### DISCUSSION TITLE
Is it okay to include `tenv` as part of a GitHub Action?

### DISCUSSION THREAD

**Discussion by rdhar** [tofuutils/tenv#259] [General]
Hi there, right off the bat, huge thanks for sharing this project, especially given how much it aids context-switching between tofu and terraform!

On that note, I was curious to get your thoughts on this: is it okay to include `tenv` as part of a GitHub Action workflow? In other words, replacing actions like setup-terraform/opentofu with `tenv` instead to install/use TF within the pipeline.

For added context, the TF-via-PR project that I maintain includes [`tenv` installation](https://github.com/DevSecTop/TF-via-PR/blob/379cb0fd793c0f7dfe2a2d8a3df5bd864ddfa3d7/action.yml#L282-L283) if the user opts for it.

From anecdotal experience, users have mentioned `tenv` makes it easier for them to dynamically switch between TF binaries seamlessly without the hassle of adding conditional-logic to install setup-terraform/opentofu actions in their workflow. I've been a fan of `tenv` for a while now, so adding first-class support for it was a must. Given both projects share the same permissive licence, I thought I'd double-check by asking the author themselves.

**Comment by ego93** [comment]
Its been hinted here https://github.com/tofuutils/tenv-github-action

**Comment by kvendingoldo** [comment]
Hi @rdhar, 
Thank you for this thread and your warm words. 
I agree, that one GitHub action for managing all tools is better. The call can looks like that:

```
- uses: tofuutils/tenv@v1
  with:
    tool_name: terraform
    tool_version: 1.5.7
```

We didn't have time to work on it, but I will try to do POC inside of the https://github.com/tofuutils/tenv-github-action; If you have time, join to the work in this repo :)

**Comment by kvendingoldo** [comment]
I kicked off the code, but haven't tested it yet.
